packages/barrier/SPECS/barrier.spec

```diff
@@ -0,0 +1,109 @@
+%global barrier_revision 8941e56f
+Summary: Keyboard and mouse sharing solution
+Name: barrier
+Version: 2.3.3
+Release: 1%{?dist}
+License: GPLv2
+Group: System Environment/Daemons
+URL: https://github.com/debauchee/barrier
+# This source is a lie, I used 43708ae0d636539f5d9627133baba7cce3ea0812 from github
+Source0: https://github.com/debauchee/barrier/archive/barrier-v%{version}.tar.gz
+Patch100: barrier.sgifixes.patch
+
+
+BuildRequires: cmake3
+#BuildRequires: avahi-compat-libdns_sd-devel
+BuildRequires: libX11-devel
+#BuildRequires: libXtst-devel
+#BuildRequires: qt5-qtbase-devel
+BuildRequires: libcurl-devel
+BuildRequires: desktop-file-utils
+BuildRequires: openssl-devel
+
+%description
+Barrier allows you to share one mouse and keyboard between multiple computers.
+Work seamlessly across Windows, macOS and Linux.
+
+%prep
+%setup -q -n %{name}-%{version}
+%patch100 -p1 -b .sgifixes
+
+%build
+%{cmake3} -DSYNERGY_VERSION_STAGE:STRING=release -DSYNERGY_REVISION:STRING=%{barrier_revision} .
+make %{?_smp_mflags}
+
+
+%install
+# no GUI for you
+# install -D -p -m 0755 bin/barrier %{buildroot}%{_bindir}/barrier
+install -D -p -m 0755 bin/barrierc %{buildroot}%{_bindir}/barrierc
+install -D -p -m 0755 bin/barriers %{buildroot}%{_bindir}/barriers
+install -D -p -m 0644 doc/barrierc.1 %{buildroot}%{_mandir}/man1/barrierc.1
+install -D -p -m 0644 doc/barriers.1 %{buildroot}%{_mandir}/man1/barriers.1
+install -D -p -m 0644 res/barrier.desktop %{buildroot}%{_datadir}/applications/barrier.desktop
+install -D -p -m 0644 res/barrier.svg %{buildroot}%{_datadir}/icons/hicolor/scalable/apps/barrier.svg
+
+cd %{buildroot}%{_bindir}
+
+mkdir -p %{buildroot}%{_datadir}/metainfo
+## Write AppStream
+cat <<END> %{buildroot}%{_datadir}/metainfo/%{name}.appdata.xml
+<?xml version="1.0" encoding="UTF-8"?>
+<!-- Copyright 2018 Ding-Yi Chen <dchen@redhat.com> -->
+<component type="desktop-application">
+  <id>%{name}</id>
+  <metadata_license>FSFAP</metadata_license>
+  <project_license>GPLv2</project_license>
+  <name>barrier</name>
+  <summary>Share mouse and keyboard between multiple computers over the network</summary>
+
+  <description>
+    <p>
+      Barrier allows you to share one mouse and keyboard between multiple computers.
+      Work seamlessly across Windows, macOS and Linux.
+    </p>
+  </description>
+
+  <launchable type="desktop-id">%{name}.desktop</launchable>
+
+  <url type="homepage">https://github.com/debauchee/barrier</url>
+
+  <provides>
+    <binary>barrier</binary>
+    <binary>barrierc</binary>
+    <binary>barriers</binary>
+  </provides>
+
+  <releases>
+    <release version="%{version}" date="2019-03-21" />
+  </releases>
+</component>
+END
+
+#desktop-file-install --delete-original \
```

Comment:
FYI we do have desktop-file-utils (it's not useful _yet_, but we might as well leave it in there).

I think we'll need a `BuildRequires: desktop-file-utils` near the top then too (it's in the fc31 version spec file, if you go that route).

Reply:
Is there any plan for that to integrate into a pane in the stock Irix application launcher or something? If it'll mostly go unused I figure on skipping it.